// svix/dist/models/customColorPalette.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomColorPaletteSerializer = void 0;
exports.CustomColorPaletteSerializer = {
    _fromJsonObject(object) {
        return {
            backgroundHover: object["backgroundHover"],
            backgroundPrimary: object["backgroundPrimary"],
            backgroundSecondary: object["backgroundSecondary"],
            buttonPrimary: object["buttonPrimary"],
            interactiveAccent: object["interactiveAccent"],
            navigationAccent: object["navigationAccent"],
            primary: object["primary"],
            textDanger: object["textDanger"],
            textPrimary: object["textPrimary"],
        };
    },
    _toJsonObject(self) {
        return {
            backgroundHover: self.backgroundHover,
            backgroundPrimary: self.backgroundPrimary,
            backgroundSecondary: self.backgroundSecondary,
            buttonPrimary: self.buttonPrimary,
            interactiveAccent: self.interactiveAccent,
            navigationAccent: self.navigationAccent,
            primary: self.primary,
            textDanger: self.textDanger,
            textPrimary: self.textPrimary,
        };
    },
};
