// svix/dist/models/endpointOauthConfigOut.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.EndpointOauthConfigOutSerializer = void 0;
const oauth2AuthMethodInOut_1 = require("./oauth2AuthMethodInOut");
const oauth2GrantTypeInOut_1 = require("./oauth2GrantTypeInOut");
exports.EndpointOauthConfigOutSerializer = {
    _fromJsonObject(object) {
        return {
            authMethod: oauth2AuthMethodInOut_1.Oauth2AuthMethodInOutSerializer._fromJsonObject(object["authMethod"]),
            clientId: object["clientId"],
            extraParams: object["extraParams"],
            grantType: oauth2GrantTypeInOut_1.Oauth2GrantTypeInOutSerializer._fromJsonObject(object["grantType"]),
            scopes: object["scopes"],
            tokenUrl: object["tokenUrl"],
        };
    },
    _toJsonObject(self) {
        return {
            authMethod: oauth2AuthMethodInOut_1.Oauth2AuthMethodInOutSerializer._toJsonObject(self.authMethod),
            clientId: self.clientId,
            extraParams: self.extraParams,
            grantType: oauth2GrantTypeInOut_1.Oauth2GrantTypeInOutSerializer._toJsonObject(self.grantType),
            scopes: self.scopes,
            tokenUrl: self.tokenUrl,
        };
    },
};
