// svix/dist/models/environmentSettingsOut.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.EnvironmentSettingsOutSerializer = void 0;
const customColorPalette_1 = require("./customColorPalette");
const customStringsOverride_1 = require("./customStringsOverride");
const customThemeOverride_1 = require("./customThemeOverride");
exports.EnvironmentSettingsOutSerializer = {
    _fromJsonObject(object) {
        return {
            colorPaletteDark: object["colorPaletteDark"]
                ? customColorPalette_1.CustomColorPaletteSerializer._fromJsonObject(object["colorPaletteDark"])
                : undefined,
            colorPaletteLight: object["colorPaletteLight"]
                ? customColorPalette_1.CustomColorPaletteSerializer._fromJsonObject(object["colorPaletteLight"])
                : undefined,
            customColor: object["customColor"],
            customFontFamily: object["customFontFamily"],
            customFontFamilyUrl: object["customFontFamilyUrl"],
            customLogoUrl: object["customLogoUrl"],
            customStringsOverride: object["customStringsOverride"]
                ? customStringsOverride_1.CustomStringsOverrideSerializer._fromJsonObject(object["customStringsOverride"])
                : undefined,
            customThemeOverride: object["customThemeOverride"]
                ? customThemeOverride_1.CustomThemeOverrideSerializer._fromJsonObject(object["customThemeOverride"])
                : undefined,
            displayName: object["displayName"],
            enableChannels: object["enableChannels"],
            enableEndpointMtlsConfig: object["enableEndpointMtlsConfig"],
            enableEndpointOauthConfig: object["enableEndpointOauthConfig"],
            enableIntegrationManagement: object["enableIntegrationManagement"],
            enableMessageStream: object["enableMessageStream"],
            enableMessageTags: object["enableMessageTags"],
            enableTransformations: object["enableTransformations"],
            showUseSvixPlay: object["showUseSvixPlay"],
            wipeSuccessfulPayload: object["wipeSuccessfulPayload"],
        };
    },
    _toJsonObject(self) {
        return {
            colorPaletteDark: self.colorPaletteDark
                ? customColorPalette_1.CustomColorPaletteSerializer._toJsonObject(self.colorPaletteDark)
                : undefined,
            colorPaletteLight: self.colorPaletteLight
                ? customColorPalette_1.CustomColorPaletteSerializer._toJsonObject(self.colorPaletteLight)
                : undefined,
            customColor: self.customColor,
            customFontFamily: self.customFontFamily,
            customFontFamilyUrl: self.customFontFamilyUrl,
            customLogoUrl: self.customLogoUrl,
            customStringsOverride: self.customStringsOverride
                ? customStringsOverride_1.CustomStringsOverrideSerializer._toJsonObject(self.customStringsOverride)
                : undefined,
            customThemeOverride: self.customThemeOverride
                ? customThemeOverride_1.CustomThemeOverrideSerializer._toJsonObject(self.customThemeOverride)
                : undefined,
            displayName: self.displayName,
            enableChannels: self.enableChannels,
            enableEndpointMtlsConfig: self.enableEndpointMtlsConfig,
            enableEndpointOauthConfig: self.enableEndpointOauthConfig,
            enableIntegrationManagement: self.enableIntegrationManagement,
            enableMessageStream: self.enableMessageStream,
            enableMessageTags: self.enableMessageTags,
            enableTransformations: self.enableTransformations,
            showUseSvixPlay: self.showUseSvixPlay,
            wipeSuccessfulPayload: self.wipeSuccessfulPayload,
        };
    },
};
