// svix/dist/models/environmentModelOut.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.EnvironmentModelOutSerializer = void 0;
const environmentRegion_1 = require("./environmentRegion");
const environmentType_1 = require("./environmentType");
exports.EnvironmentModelOutSerializer = {
    _fromJsonObject(object) {
        return {
            createdAt: new Date(object["createdAt"]),
            id: object["id"],
            name: object["name"],
            region: environmentRegion_1.EnvironmentRegionSerializer._fromJsonObject(object["region"]),
            type: environmentType_1.EnvironmentTypeSerializer._fromJsonObject(object["type"]),
            updatedAt: new Date(object["updatedAt"]),
        };
    },
    _toJsonObject(self) {
        return {
            createdAt: self.createdAt,
            id: self.id,
            name: self.name,
            region: environmentRegion_1.EnvironmentRegionSerializer._toJsonObject(self.region),
            type: environmentType_1.EnvironmentTypeSerializer._toJsonObject(self.type),
            updatedAt: self.updatedAt,
        };
    },
};
