// svix/dist/models/settingsInternalOut.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SettingsInternalOutSerializer = void 0;
const customColorPalette_1 = require("./customColorPalette");
const customStringsOverride_1 = require("./customStringsOverride");
const customThemeOverride_1 = require("./customThemeOverride");
exports.SettingsInternalOutSerializer = {
    _fromJsonObject(object) {
        return {
            colorPaletteDark: object["colorPaletteDark"]
                ? customColorPalette_1.CustomColorPaletteSerializer._fromJsonObject(object["colorPaletteDark"])
                : undefined,
            colorPaletteLight: object["colorPaletteLight"]
                ? customColorPalette_1.CustomColorPaletteSerializer._fromJsonObject(object["colorPaletteLight"])
                : undefined,
            customBaseFontSize: object["customBaseFontSize"],
            customColor: object["customColor"],
            customFontFamily: object["customFontFamily"],
            customFontFamilyUrl: object["customFontFamilyUrl"],
            customLogoUrl: object["customLogoUrl"],
            customStringsOverride: object["customStringsOverride"]
                ? customStringsOverride_1.CustomStringsOverrideSerializer._fromJsonObject(object["customStringsOverride"])
                : undefined,
            customThemeOverride: object["customThemeOverride"]
                ? customThemeOverride_1.CustomThemeOverrideSerializer._fromJsonObject(object["customThemeOverride"])
                : undefined,
            disableEndpointOnFailure: object["disableEndpointOnFailure"],
            displayName: object["displayName"],
            enableChannels: object["enableChannels"],
            enableEndpointMtlsConfig: object["enableEndpointMtlsConfig"],
            enableEndpointOauthConfig: object["enableEndpointOauthConfig"],
            enableIntegrationManagement: object["enableIntegrationManagement"],
            enableMessageStream: object["enableMessageStream"],
            enableMsgAtmptLog: object["enableMsgAtmptLog"],
            enableOtlp: object["enableOtlp"],
            enableTransformations: object["enableTransformations"],
            enforceHttps: object["enforceHttps"],
            eventCatalogPublished: object["eventCatalogPublished"],
            readOnly: object["readOnly"],
            requireEndpointChannel: object["requireEndpointChannel"],
            retryPolicy: object["retryPolicy"],
            showUseSvixPlay: object["showUseSvixPlay"],
            whitelabelHeaders: object["whitelabelHeaders"],
            wipeSuccessfulPayload: object["wipeSuccessfulPayload"],
        };
    },
    _toJsonObject(self) {
        return {
            colorPaletteDark: self.colorPaletteDark
                ? customColorPalette_1.CustomColorPaletteSerializer._toJsonObject(self.colorPaletteDark)
                : undefined,
            colorPaletteLight: self.colorPaletteLight
                ? customColorPalette_1.CustomColorPaletteSerializer._toJsonObject(self.colorPaletteLight)
                : undefined,
            customBaseFontSize: self.customBaseFontSize,
            customColor: self.customColor,
            customFontFamily: self.customFontFamily,
            customFontFamilyUrl: self.customFontFamilyUrl,
            customLogoUrl: self.customLogoUrl,
            customStringsOverride: self.customStringsOverride
                ? customStringsOverride_1.CustomStringsOverrideSerializer._toJsonObject(self.customStringsOverride)
                : undefined,
            customThemeOverride: self.customThemeOverride
                ? customThemeOverride_1.CustomThemeOverrideSerializer._toJsonObject(self.customThemeOverride)
                : undefined,
            disableEndpointOnFailure: self.disableEndpointOnFailure,
            displayName: self.displayName,
            enableChannels: self.enableChannels,
            enableEndpointMtlsConfig: self.enableEndpointMtlsConfig,
            enableEndpointOauthConfig: self.enableEndpointOauthConfig,
            enableIntegrationManagement: self.enableIntegrationManagement,
            enableMessageStream: self.enableMessageStream,
            enableMsgAtmptLog: self.enableMsgAtmptLog,
            enableOtlp: self.enableOtlp,
            enableTransformations: self.enableTransformations,
            enforceHttps: self.enforceHttps,
            eventCatalogPublished: self.eventCatalogPublished,
            readOnly: self.readOnly,
            requireEndpointChannel: self.requireEndpointChannel,
            retryPolicy: self.retryPolicy,
            showUseSvixPlay: self.showUseSvixPlay,
            whitelabelHeaders: self.whitelabelHeaders,
            wipeSuccessfulPayload: self.wipeSuccessfulPayload,
        };
    },
};
